ccd-definition/CaseEventToComplexTypes/create-c21/C21Order-gatekeeping.json

```diff
@@ -0,0 +1,73 @@
+[
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "temporaryC21Order",
+    "ListElementCode": "orderTitle",
+    "EventElementLabel": "Order title",
+    "FieldDisplayOrder": 1,
+    "DisplayContext": "OPTIONAL"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "temporaryC21Order",
+    "ListElementCode": "orderDetails",
+    "EventElementLabel": "Order details",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderCheckAnswer",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "temporaryC21Order",
+    "ListElementCode": "c21OrderDocument",
```

Comment:
hmm I have seen this is in test... but I would like to make some comments.

c21OrderDocument seems over kill here... temporaryC21Order.c21OrderDocument is how this will be rendered in ccd. I have the context of c21Order from the parent. In my mind the listElementCode here should just be document

Reply:
similarly should I change orderTitle and orderDetails to just be title and details? might be more consistent - or I change this to orderDocument and then they all are prefixed with 'order'

Reply:
I am not 100% sure of the context of this story, but I think orderTitle and orderDetails make sense... because they are the title of the order and details associated with that c21. The document however, is that c21. Is that correct?

Reply:
yes essentially the document is the C21 (I think) - Ill change document list element code

Reply:
hmm no document as a list element code doesn't sound correct, I don't think that is what I am getting at.

In CCD I imagine one object - a C21 Order, with orderTitle, orderDetails, judgeAndLegalAdvisor, and a document. I imagine you will use the c21Order object as a display item, i.e it will sit in the UI and gather data. You will then use a callback to take the individual c21Order and place into a collection of c21 order. So... the data will be stored in a collection of c21Order, and this will then be displayed in tabs as the list. 

Does my thinking here make sense?

Reply:
currently you have two objects and then a collection of one of those objects. You display one (which I do not think is needed), you add it all together in another object (that has a doc field and judgeAndLegalAdvisor) (... this is the one I want you to use), it is then placed into a collection

Reply:
doing the above... I then imagine the java code in the aboutToSubmit to look something like
```
c21Order.toBuilder()
.judgeAndLegalAdvisor(caseData.getJALA)
.document(where ever the doc is from)
.build() 
```
make sense?

Reply:
then something like
```
caseData.getC21Orders().add(Element<c21Order>.builder()
    .id(whatever)
    .value(the c21 up above)
.build();
```

then

```
caseDetails.getData().put("c21Orders", caseData.getc21Orders);
```

Reply:
where caseData is

```
caseData: {
    c21Order: {
        order stuff ...
    },
    c21Orders: [{
        c21Order1,
        c21Order2
    }],
}
```

Reply:
yep your thinking makes total sense - my only concern is I'm still going to need the judge object for the 2nd page anyway (right?), and that I don't want to display the order details of the C21 order in the tab. but I don't know how if we can hide a part of the collection (specifically the order details of a C21 order) in the tab view - so is this refactor a good idea? but yes I could in theory add the judge from 2nd page into the C21Order and then have our collection of C21Orders (which contain title, details, document, judge stuff, and the date/time of creation). i tried this initially, it got messy